### PR TITLE
feat: Display the reason for application pod restarts on the Devtron UI

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Readme icon inpod restart

Fixes https://github.com/devtron-labs/devtron/issues/3792

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 1]
```diff
@@ -416,6 +421,9 @@ const NodeComponent = ({
                         {params.nodeType === NodeType.Pod.toLowerCase() && (
                             <div data-testid="pod-restart-count" className="flex left col-1 pt-9 pb-9">
                                 {node.kind !== 'Containers' && getPodRestartCount(node)}
+                                {PodRestartIcon && (
+                                    <PodRestartIcon clusterId={clusterId} name={node.name} namespace={node.namespace} />
+                                )}
```
Comment (eshankvaish) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:426]: pls check if a gap is required here

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/v2/appDetails/appDetails.type.ts [hunk 2]
```diff
@@ -525,6 +526,7 @@ export interface K8ResourceComponentProps {
     monitoringTools: OptionTypeWithIcon[]
     isExternalApp?: boolean
     isDevtronApp?: boolean
+    clusterId?: number
```
Comment (eshankvaish) [inline src/components/v2/appDetails/appDetails.type.ts:529]: we can use explicit number | null instead. and get rid of the optional

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/charts/discoverChartDetail/DiscoverChartDetails.tsx [hunk 3]
```diff
@@ -412,6 +412,11 @@ function isReadmeInputCheckbox(text: string) {
     }
     return false
 }
+
+/**
+ * 
+ * deprecated function is used in common component
```
Comment (eshankvaish) [inline src/components/charts/discoverChartDetail/DiscoverChartDetails.tsx:415]:
```suggestion
 * @deprecated function is used in common component
```

File: package.json [hunk 4]
```diff
@@ -80,6 +80,8 @@
     "devDependencies": {
         "@playwright/test": "^1.32.1",
         "@sentry/cli": "^2.2.0",
+        "marked": "4.3.0",
+        "dompurify": "^3.0.2",
```
Comment (eshankvaish) [inline package.json:84]: move to deps

File: src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx [hunk 5]
```diff
@@ -355,6 +368,9 @@ export const K8SResourceList = ({
                 renderCallBackSync={renderCallBackSync}
             />
             {resourceListLoader ? <Progressing pageLoader /> : renderList()}
+            {PodRestart && (
+                <PodRestart clusterId={clusterId} name={name ?? 'demo2-vishu-env-5ddcb8f64b-lgrqk'} namespace={namespace ?? 'non-prod'} />
```
Comment (eshankvaish) [inline src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx:372]: pls revert when testing is done

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 6]
```diff
@@ -524,6 +525,13 @@ const NodeComponent = ({
                     )}
                 </div>
             )}
+            {PodRestart && (
+                <PodRestart
+                    clusterId={clusterId}
+                    name={params.name ?? 'demo2-vishu-env-5ddcb8f64b-lgrqk'}
```
Comment (eshankvaish) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:531]: .

File: src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx (hunk 5 above)
Comment (eshankvaish) [reply src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx:372]: also pls use useParams inside the components

File: src/components/v2/appDetails/appDetails.type.ts (hunk 2 above)
Comment (eshankvaish) [reply src/components/v2/appDetails/appDetails.type.ts:529]: remove the optional check

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (shivani170) [reply package.json:84]: Already in dependencies, removed from devdep

File: src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx (hunk 5 above)
Comment (shivani170) [reply src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx:372]: Done

File: src/components/v2/appDetails/NodeTreeDetailTab.tsx [hunk 7]
```diff
@@ -17,6 +17,7 @@ const NodeTreeDetailTab = ({
     monitoringTools,
     isDevtronApp = false,
     isExternalApp,
+    clusterId,
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/NodeTreeDetailTab.tsx:20]: No need to introduce new prop. we are anyways getting the whole appDetails here

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Comment by shivani170** [comment]
Merged in demo/resource-browser

**Pull request closed without merging**